sdk/src/jmh/java/io/opentelemetry/sdk/metrics/MetricsTestOperationBuilder.java

```diff
@@ -0,0 +1,173 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import com.google.errorprone.annotations.Immutable;
+import io.opentelemetry.metrics.DoubleCounter;
+import io.opentelemetry.metrics.DoubleMeasure;
+import io.opentelemetry.metrics.LongCounter;
+import io.opentelemetry.metrics.LongMeasure;
+import io.opentelemetry.metrics.Meter;
+
+/**
+ * This enum allows for iteration over all of the operations that we want to benchmark. To ensure
+ * that the enum cannot change state, each enum holds a builder function- passing a meter in will
+ * return a wrapper for both bound and unbound versions of that operation which can then be used in
+ * a benchmark.
+ */
+public enum MetricsTestOperationBuilder {
+  LongCounterAdd(
+      new OperationBuilder<LongCounter, LongCounter.BoundLongCounter>() {
+        @Override
+        public Operation<LongCounter, LongCounter.BoundLongCounter> build(Meter meter) {
+          Operation<LongCounter, LongCounter.BoundLongCounter> op =
+              new Operation<LongCounter, LongCounter.BoundLongCounter>() {
+                @Override
+                void perform(String... args) {
+                  metric.add(5L, args);
+                }
+
+                @Override
+                void performBound() {
+                  boundMetric.add(5L);
+                }
+
+                @Override
+                protected void initialize(Meter meter) {
+                  metric = meter.longCounterBuilder("long_counter").build();
+                  boundMetric =
+                      meter.longCounterBuilder("bound_long_counter").build().bind("KEY", "VALUE");
+                }
+              };
+          op.initialize(meter);
+          return op;
+        }
+      }),
+  DoubleCounterAdd(
+      new OperationBuilder<DoubleCounter, DoubleCounter.BoundDoubleCounter>() {
+        @Override
+        public Operation<DoubleCounter, DoubleCounter.BoundDoubleCounter> build(Meter meter) {
+          Operation<DoubleCounter, DoubleCounter.BoundDoubleCounter> op =
+              new Operation<DoubleCounter, DoubleCounter.BoundDoubleCounter>() {
+                @Override
+                void perform(String... args) {
+                  metric.add(5.0d, args);
+                }
+
+                @Override
+                void performBound() {
+                  boundMetric.add(5.0d);
+                }
+
+                @Override
+                protected void initialize(Meter meter) {
+                  metric = meter.doubleCounterBuilder("double_counter").build();
+                  boundMetric =
+                      meter
+                          .doubleCounterBuilder("bound_double_counter")
+                          .build()
+                          .bind("KEY", "VALUE");
+                }
+              };
+          op.initialize(meter);
```

Comment:
It seems that this is a common pattern to call initialize as the fist thing after the ctor is called. Do we really need an `initialize` method or we can just move the code and get rid of that method?

Reply:
We need `initialize(meter)`, because we can't have a constructor in an abstract class, and we need somewhere to build the correct metric/boundMetric set for the test. Code standards don't allow for calling an overridden method from a superclass constructor, but I can move this into a superclass `build(Meter meter)` that will clean this up a bit. The other alternative is to move all of these Operations into a named class and use constructors, but I think that will be less clear.

Reply:
Or I missed an obvious way to restructure this by passing the meter and bound meter into the constructor.  Oops.